hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/upgrade/UpgradeDowngrade.java

```diff
@@ -90,14 +90,34 @@ public static boolean needsUpgradeOrDowngrade(HoodieTableMetaClient metaClient,
 
   public static boolean needsUpgrade(HoodieTableMetaClient metaClient, HoodieWriteConfig config, HoodieTableVersion toWriteVersion) {
     HoodieTableVersion fromTableVersion = metaClient.getTableConfig().getTableVersion();
-    // If table version is less than SIX, then we need to upgrade to SIX first before upgrading to any other version, irrespective of autoUpgrade flag
+    if (fromTableVersion.versionCode() >= toWriteVersion.versionCode()) {
+      // if the table version is greater than or equal to the write version, then this is not an upgrade
+      LOG.warn("Table version {} is greater than or equal to write version {}. No upgrade needed", fromTableVersion, toWriteVersion);
```

Comment:
In this case and autoUpgrade is false and `fromTableVersion.versionCode() > toWriteVersion.versionCode()`, should an error be thrown given the mismatch (or does the code automatically elevate the write table version to be the same as the existing table version)?  Could a test case like this be added at the write client or data source write level?